classes/Product.php

```diff
@@ -7087,7 +7087,9 @@ public function setWsPositionInCategory($position)
         // result is indexed by recordset order and not position. positions start at index 1 so we need an empty element
         array_unshift($result, null);
         foreach ($result as &$value) {
-            $value = $value['id_product'];
+            if($value != null) {
```

Comment:
Just move the array unshift after the `foreach` would avoid getting for logic in here